### PR TITLE
Refactor ViTDet backbone and simple feature pyramid

### PR DESCRIPTION
**Pull request by tuofeilunhifi**
# Motivation
Align with ViTDet of detectron2

# Modification
Refactor vitdet backbone and simple feature pyramid.

# illustrate
https://github.com/facebookresearch/detectron2/blob/main/MODEL_ZOO.md
https://github.com/open-mmlab/mmdetection/tree/master/configs/mask_rcnn
https://github.com/open-mmlab/mmdetection/tree/master/configs/cascade_rcnn

Since we use the mask rcnn of mmdet as the basline, but the (R50-FPN mask rcnn 1x) of d2 is 0.4-0.6 higher than that of mmdet, theoretically the upper limit of the ViTDet-base we reproduce should be under 51 mAP.
(In addition，R50-FPN cascade mask rcnn 1x of d2 is 1.7-1.8 higher than that of mmdet)

# Results
1. The vitdet backbone implemented by d2 is about 20% faster than the vitdet backbone originally reproduced by easycv.
2. 50.57 -> 50.65

### CODE REVIEW

File: configs/detection/vitdet/lsj_coco_detection.py [hunk 1]
```diff
@@ -101,13 +101,16 @@
     pipeline=test_pipeline)
 
 data = dict(
-    imgs_per_gpu=1, workers_per_gpu=2, train=train_dataset, val=val_dataset)
+    imgs_per_gpu=1, workers_per_gpu=2, train=train_dataset, val=val_dataset
+)  # 64(total batch size) = 4 (batch size/per gpu) x 8 (gpu num) x 2(node)
```
Comment (Cathy0908) [inline configs/detection/vitdet/lsj_coco_detection.py:105]: 实际参数与备注不同，imgs_per_gpu=1,备注写的4

File: configs/detection/vitdet/lsj_coco_detection.py [hunk 2]
```diff
@@ -101,13 +101,16 @@
...
 
 # evaluation
-eval_config = dict(interval=1, gpu_collect=False)
+eval_config = dict(initial=True, interval=1, gpu_collect=False)
```
Comment (Cathy0908) [inline configs/detection/vitdet/lsj_coco_detection.py:108]: initial=False

File: configs/detection/vitdet/vitdet_schedule_100e.py [hunk 3]
```diff
@@ -1,26 +1,29 @@
 _base_ = 'configs/base.py'
```
Comment (Cathy0908) [inline configs/detection/vitdet/vitdet_schedule_100e.py:1]: What does the name of this file mean vitdet_schedule_100e.py？

Comment (tuofeilunhifi) [reply configs/detection/vitdet/vitdet_schedule_100e.py:1]: vitdet training plan，total 100 epoch

File: configs/detection/vitdet/lsj_coco_detection.py (hunk 1 above)
Comment (tuofeilunhifi) [reply configs/detection/vitdet/lsj_coco_detection.py:105]: done

File: configs/detection/vitdet/lsj_coco_detection.py (hunk 2 above)
Comment (tuofeilunhifi) [reply configs/detection/vitdet/lsj_coco_detection.py]: done

File: easycv/core/optimizer/layer_decay_optimizer_constructor.py [hunk 4]
```diff
@@ -7,17 +7,25 @@
...
-    elif var_name.startswith('backbone.patch_embed'):
-        return 0
-    elif var_name.startswith('backbone.blocks'):
-        layer_id = int(var_name.split('.')[2])
-        return layer_id + 1
-    else:
-        return num_max_layer - 1
+def get_vit_lr_decay_rate(name, lr_decay_rate=1.0, num_layers=12):
+    """
+    Calculate lr decay rate for different ViT blocks.
```
Comment (wenmengzhou) [inline easycv/core/optimizer/layer_decay_optimizer_constructor.py:12]: 描述一下适用于哪些训练场景，或者有什么ref的paper

Comment (tuofeilunhifi) [reply easycv/core/optimizer/layer_decay_optimizer_constructor.py:13]: done

Comment (tuofeilunhifi) [reply easycv/core/optimizer/layer_decay_optimizer_constructor.py:10]: done

**Review by wenmengzhou — APPROVED** [review]
